### PR TITLE
Add TimeValidity emulator test for different protocol versions

### PR DESCRIPTION
**Pull request by james-iohk**
Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [ ] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-contract/test/Spec/TimeValidity.hs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    "tx valid time interval is supported in protocol v6+"
+    (assertValidatedTransactionCount 2)
+    (void trace)
+
+deadline :: POSIXTime
+deadline = 1596059092000 -- (milliseconds) transaction's valid range must be after this
+
+{-# INLINEABLE mkValidator #-}
+mkValidator :: P.POSIXTime -> () -> () -> P.ScriptContext -> Bool
+mkValidator dl _ _ ctx = (from dl `contains` range) || P.traceError "Invalid range"
```
Comment (sjoerdvisscher) [inline plutus-contract/test/Spec/TimeValidity.hs:93]: Nice!

**Review by sjoerdvisscher — APPROVED** [review]

File: plutus-contract/test/Spec/TimeValidity.hs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    let tx1 = Constraints.mustPayToOtherScript valHash unitDatum $ Ada.lovelaceValueOf 25000000
+    ledgerTx1 <- submitTx tx1
+    awaitTxConfirmed $ Tx.getCardanoTxId ledgerTx1
+    utxos <- utxosAt scrAddress
+    let orefs = fst <$> Map.toList utxos
+        lookups =
+            Constraints.otherScript (validatorScript deadline)
+            <> Constraints.unspentOutputs utxos
+        tx2 =
+            mconcat [Constraints.mustSpendScriptOutput oref unitRedeemer | oref <- orefs]
```
Comment (koslambrou) [inline plutus-contract/test/Spec/TimeValidity.hs:56]: Use `foldMap` here

File: plutus-contract/test/Spec/TimeValidity.hs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+typedValidator :: POSIXTime -> Scripts.TypedValidator UnitTest
+typedValidator = Scripts.mkTypedValidatorParam @UnitTest
+    $$(PlutusTx.compile [||mkValidator||])
+    $$(PlutusTx.compile [|| wrap ||])
+    where
+        wrap = Scripts.mkUntypedValidator
+
+validatorScript :: P.POSIXTime -> Validator
+validatorScript = Scripts.validatorScript . typedValidator
```
Comment (koslambrou) [inline plutus-contract/test/Spec/TimeValidity.hs:113]: You can simply all of that code to something like:

```
validatorScript :: Plutus.Validator
validatorScript = Plutus.mkValidatorScript
    $$(PlutusTx.compile [|| wrap ||])
 where
     wrap = Scripts.mkUntypedValidator mkValidator

```

Comment (koslambrou) [reply plutus-contract/test/Spec/TimeValidity.hs:113]: Unless you plan to use the `TypedValidator` with the constraints library.

Comment (james-iohk) [reply plutus-contract/test/Spec/TimeValidity.hs:114]: Am now using `typedValidatorLookups`

**Review by koslambrou — APPROVED** [review]

**Pull request merged**